benchmarks/distributed/rpc/ps/launcher.py

```diff
@@ -0,0 +1,372 @@
+import argparse
+import copy
+import json
+import os
+from pathlib import Path
+
+import torch.distributed.rpc as rpc
+import torch.multiprocessing as mp
+from torch.distributed.rpc import TensorPipeRpcBackendOptions
+from torch.utils.data import DataLoader
+
+from benchmark_class_helper import (get_benchmark_data_map,
+                                    get_benchmark_model_map,
+                                    get_benchmark_ps_map,
+                                    get_benchmark_trainer_map)
+from BenchmarkConfigurations import BenchmarkConfigurations
+from metrics.ProcessedMetricsPrinter import ProcessedMetricsPrinter
+
+USE_CUDA_RPC = "use_cuda_rpc"
+
+
+def get_name(rank, trainer_count, parameter_server_count):
+    if rank < trainer_count:
+        return "trainer{}".format(rank)
+    elif rank < (trainer_count + parameter_server_count):
+        return "ps{}".format(rank)
+    else:
+        return "master"
+
+
+def get_parameter_server_rank(rank, configurations):
+    if configurations.parameter_server_count < 1:
+        return None
+    # rank mod parameter server count to get parameter server number
+    # add trainer_count to get parameter server rank
+    rank_mod_ps_count = rank % configurations.parameter_server_count
+    return rank_mod_ps_count + configurations.trainer_count
+
+
+def print_metrics(metrics_printer, rank_metrics_list, name):
```

Comment:
curious, why do we need an additional helper function instead of putting this to the metrics printer class implementation?

Reply:
Ill move it to the metrics printer class.